spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceObject.java

```diff
@@ -205,20 +223,32 @@ public String getDescription() {
 
 	@Override
 	public InputStream getInputStream() throws IOException {
-		return Channels.newInputStream(throwExceptionForNullBlob(getGoogleStorageObject()).reader());
+		return Channels.newInputStream(throwExceptionForNullBlob(getGcsBlob()).reader());
 	}
 
 	@Override
 	public boolean isWritable() {
 		return true;
 	}
 
+	/**
+	 * Returns the output stream for a Google Cloud Storage file.
+	 * @return the object's output stream or {@code null} if the object doesn't exist and cannot be
+	 * created
+	 * @throws IOException
+	 */
 	@Override
 	public OutputStream getOutputStream() throws IOException {
-		Blob blob = getGoogleStorageObject();
-		if ((blob == null || !blob.exists()) && this.createBlobIfNotExists) {
-			blob = createBlob();
+		Blob blob = getGcsBlob();
+
+		if (blob == null || !blob.exists()) {
+			if (!this.createBlobIfNotExists) {
+				return null;
```

Comment:
I would skip this whole if statement. Just let the final return statement throw an exception for non-existing blob.

Reply:
I don't think we should do that because of #165. Unless I'm missing anything?

Thoughts?

Reply:
The issue I'm pointing out here is the same as above. Communicating an error condition by returning `null`. It's better to throw an exception IMO, just like it was before your change.